interface/main/backup.php

```diff
@@ -604,13 +953,42 @@ function gzopen($filename, $mode, $use_include_path = 0)
     " -p" . escapeshellarg($sqlconf["pass"]) .
     " -h " . escapeshellarg($sqlconf["host"]) .
     " --port=" . escapeshellarg($sqlconf["port"]) .
-    " --hex-blob --opt --quote-names -r " . escapeshellarg($BACKUP_EVENTLOG_FILE) . " $mysql_ssl " .
+    " --hex-blob --opt --quote-names --no-tablespaces -r " . escapeshellarg($BACKUP_EVENTLOG_FILE) . " $mysql_ssl " .
     escapeshellarg($sqlconf["dbase"]) . " --tables log_comment_encrypt_backup log_backup api_log_backup";
 # Set Eventlog Flag when it is done
     $eventlog = 1;
 // 301 If ends here.
 }
 
+if ($form_step == 401) {
+    echo "<p><b>&nbsp;" . xlt('Download or Delete Old Log Entries') . ":</b></p>";
+    $tmprow = sqlQuery("SELECT COUNT(*) AS count, MIN(date) AS date FROM log");
+    echo "<p>&nbsp;" . xlt('The log has') . ' ' . $tmprow['count'] . ' '  .
+        xlt('entries with the oldest dated') . ' ' . $tmprow['date'] . ".</p>";
+    // Default end date is end of year 2 years ago, ensuring 1 full year of log remaining.
+    $end_date = fixDate($_POST['form_end_date'] ?? '', (date('Y') - 2) . '-12-31');
+    echo "<p>&nbsp;" . xlt('Select an end date. Entries after this date will not be downloaded or deleted.') . " ";
+    echo "<input type='text' class='datepicker' name='form_end_date' id='form_end_date' size='10' " .
+        "value='" . attr(oeFormatShortDate($end_date)) . "' " .
+        "onkeyup='datekeyup(this,mypcc)' onblur='dateblur(this,mypcc)' title='End date yyyy-mm-dd' />";
+    echo "</p>\n";
+    echo "<p><input type='button' onclick='export_submit(402)' value='" . xla('Download Log Entries as Zipped CSV') . "' />&nbsp;\n";
+    echo "<input type='button' onclick='export_submit(405)' value='" . xla('Delete Log Entries') . "' /></p>\n";
+}
+
+if ($form_step == 405) {
+    // Process log delete, then optimize to reclaim the file space.
+    $end_date = fixDate($_POST['form_end_date'], '');
+    if ($end_date) {
+        sqlStatement("DELETE FROM log WHERE date <= ?", array("$end_date 23:59:59"));
+        sqlStatement("OPTIMIZE TABLE log");
```

Comment:
deleting log is more complicated since there are other tables that depend on it (or else log viewing/linking and log tampering checking will all break). See the 'Create Eventlog Backup' feature in this script for all that is entailed when deleting the log (aka form_step 301).

Reply:
OK does this look reasonable? I'm not sure if any other tables are dependent.

`    sqlStatement(
        "DELETE log, lce, al FROM log " .
        "LEFT JOIN log_comment_encrypt AS lce ON lce.log_id = log.id " .
        "LEFT JOIN api_log AS al ON al.log_id = log.id " .
        "WHERE log.date <= ?",
        array("$end_date 23:59:59")
    );
`

Reply:
Sorry about the lack of line breaks above.

Reply:
looks good to me (as long as working on your testing)